app/services/spam_report_service.rb

```diff
@@ -1,14 +1,11 @@
 class SpamReportService < ApplicationService
-  attr_reader :delivery_attempt
+  attr_reader :subscriber
 
-  def initialize(delivery_attempt)
-    @delivery_attempt = delivery_attempt
+  def initialize(subscriber)
+    @subscriber = subscriber
   end
 
   def call
-    subscriber_id = delivery_attempt.email.subscriber_id
-    subscriber = Subscriber.find(subscriber_id)
     UnsubscribeAllService.call(subscriber, :marked_as_spam)
```

Comment:
I don't think there is much value in keeping this as a class if it only has one method.